src/Microsoft.AspNet.Diagnostics.Abstractions/DiagnosticMessage.cs

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNet.Diagnostics
+{
+    /// <summary>
+    /// A single diagnostic message.
+    /// </summary>
+    public class DiagnosticMessage
+    {
+        public DiagnosticMessage(
```

Comment:
I removed some of the properties from this - `DiagnosticMessageSeverity`, `ErrorCode` and `object source`. Not sure if there's value in keeping the first - the latter two were used in tooling scenarios.